### ISSUE TITLE
Let users further customize page css

### ISSUE THREAD

**Issue by ford04** [m-khvoinitsky/dark-background-light-text-extension#194]
I would like to further customize pages based on one of this extension's presets.

Ideally if the extension is configured and enabled for a specific page with a preset like "default", you could add custom CSS styles to it which are appended after "default" preset. If the extension is disabled for the page, these styles also should not be loaded (because they are built on top of a dark theme).

An example would be github.com itself. I think , the dark "default " theme for it is very cool, but I dislike these nasty white border grid lines 🙂

![image](https://user-images.githubusercontent.com/16099458/44215659-568a1100-a173-11e8-8834-d82900a1a368.png)

Are you planning to incorporate such a feature ? Or can you recommend an alternative solution?

Greetings

**Comment by jkemp814** [comment]
I agree with this request also. The white border lines are too much contrast. If the border has to be displayed it should be grey, in my opinion.

**Comment by m-khvoinitsky** [comment]
See #125 for borders.

>  Let users further customize page css

I don't want to do this. There is Stylus which does it better. However:
> If the extension is disabled for the page, these styles also should not be loaded

This of course won't work with Stylus, probably it makes sense to add some attribute to `<html>` to indicate which method currently is in use so users would be able to use it in selectors.

**Comment by hklene** [comment]
I was trying https://www.w3schools.com/jsref/jsref_getcomputedstyle.asp to find out if the background color is black or white. But it's always reported as plain white and does not reflect the status of this addon. Is this some kind of security feature so servers cannot identify and trac users by checking for effects of installed addons?

Is there any guideline, how to design / colorize a website to be easily compatible with your addon and look good by default?

**Comment by m-khvoinitsky** [comment]
> I was trying https://www.w3schools.com/jsref/jsref_getcomputedstyle.asp to find out if the background color is black or white. But it's always reported as plain white and does not reflect the status of this addon. Is this some kind of security feature so servers cannot identify and trac users by checking for effects of installed addons?

Well, it shouldn't. I would actually happy do do something like this (preventing website from knowing whether user uses my add-on or not and knowing theirs preferences) to eliminate tracking and bugs like #153 but I can't. What exact code are you using? And what add-on's method you're trying to detect?

> Is there any guideline, how to design / colorize a website to be easily compatible with your addon and look good by default?

For now, no. Probably, it should be. Add-on's approach always was trying to adapt to existing web, not vice-versa. Some hints from the top of my head:
* use `<img>`s for meaningful pictures, CSS's `backgound-image` for pure decoration
* use icons that are recognizable on any background
* do not generate stylesheets dynamically.
* whatever you do, test for compatibility.
